torch_geometric/loader/temporal_dataloader.py

```diff
@@ -26,13 +29,44 @@ def __init__(self, data: TemporalData, batch_size: int = 1, **kwargs):
 
         self.data = data
         self.events_per_batch = batch_size
-
+        self.device = device
+        self.negative_sampling = negative_sampling
+        if self.negative_sampling:
+            # Ensure to only sample actual destination nodes as negatives.
+            self.min_dst_idx, self.max_dst_idx = int(self.data.dst.min()), int(
+                self.data.dst.max())
+        self.neighbor_loader = LastNeighborLoader(data.num_nodes,
+                                                  size=neighbor_loader_size,
+                                                  device=self.device)
+        # Helper vector to map global node indices to local ones.
+        self.assoc = torch.empty(data.num_nodes, dtype=torch.long,
+                                 device=self.device)
         if kwargs.get('drop_last', False) and len(data) % batch_size != 0:
             arange = range(0, len(data) - batch_size, batch_size)
         else:
             arange = range(0, len(data), batch_size)
 
         super().__init__(arange, 1, shuffle=False, collate_fn=self, **kwargs)
 
+    def _get_iterator(self):
+        self.neighbor_loader.reset_state()  # Start with an empty graph.
```

Comment:
As mentioned earlier: You don't want to start with an empty graph for validation/testing.